src/studio/src/designer/frontend/packages/schema-editor/src/components/SchemaInspector/ItemRestrictionsTab.tsx

```diff
@@ -130,70 +72,61 @@ export const ItemRestrictionsTab = ({
     event.preventDefault();
     dispatchAddEnum();
   };
+
+  const t = (key: string) => getTranslation(key, language);
+  const restrictionProps: RestrictionItemProps = {
+    restrictions: item.restrictions ?? [],
+    readonly: item.$ref !== undefined,
+    path: item.path,
+    onChangeRestrictionValue,
+    language,
+  };
   return (
     <Grid container spacing={1} className={classes.gridContainer}>
       <Grid item xs={12}>
         <FormControlLabel
           className={classes.header}
-          control={
-            <Checkbox
-              checked={selectedItem.isRequired}
-              onChange={handleRequiredChanged}
-              name='checkedRequired'
-            />
-          }
-          label={getTranslation('required', language)}
+          control={<Checkbox checked={item.isRequired} onChange={handleRequiredChanged} name='checkedRequired' />}
+          label={t('required')}
         />
       </Grid>
-      {selectedItem.$ref === undefined && (
-        <>
-          <Grid item xs={12}>
-            <Divider />
-          </Grid>
-          <Grid item xs={4}>
-            <p>{getTranslation('keyword', language)}</p>
-          </Grid>
-          <Grid item xs={1} />
-          <Grid item xs={7}>
-            <p>{getTranslation('value', language)}</p>
-          </Grid>
-          {renderItemRestrictions(selectedItem)}
-          <IconButton
-            id='add-restriction-button'
-            aria-label={getTranslation('add_restriction', language)}
-            onClick={onAddRestrictionClick}
-          >
-            <i className='fa fa-plus' />
-            {getTranslation('add_restriction', language)}
-          </IconButton>
-        </>
+      {item.$ref === undefined && (
+        <Grid item xs={12}>
+          {
+            {
+              [FieldType.Array]: <ArrayRestrictions {...restrictionProps} />,
+              [FieldType.Boolean]: <BooleanRestrictions {...restrictionProps} />,
+              [FieldType.Integer]: <NumberRestrictions {...restrictionProps} />,
+              [FieldType.Number]: <NumberRestrictions {...restrictionProps} />,
+              [FieldType.Object]: <ObjectRestrictions {...restrictionProps} />,
+              [FieldType.String]: <StringRestrictions {...restrictionProps} />,
+              [FieldType.Null]: undefined,
+              default: undefined,
+            }[item.type ?? 'default']
+          }
+        </Grid>
```

Comment:
Trenger hver type å ha sin egen komponent på denne måten? Med det får vi jo fem filer som er nesten identiske.

Reply:
Det er for neste steg i den refaktoreringen, da hver restriksjon vil ha litt ulike fremtoning.